### DISCUSSION TITLE
astroid-package-v3.0.15.Framework-Template.zip install error: An error has occurred.  0 define() expects at least 2 arguments, 1 given

### DISCUSSION THREAD

**Discussion by Whichcraft** [templaza/astroid-framework#714] [Q&A]
Hi folks. I've previously used the old astroid framework from jd boston on joomla 3.x and have now upgraded to Joomla 5.1.1 but am not able to install the astroid-package-v3.0.15.Framework-Template.zip, while the framework (astroid-framework-3.0.15.zip) installed just fine. Any ideas on what I am doing wrong or ho to fix it?

 define() expects at least 2 arguments, 1 given

# | Function | Location
-- | -- | --
1 | () | JROOT/tmp/install_6682de7cc1d30/astroid-package-v3.0.15.Framework-Template/pkg_astroid.php:64
2 | define() | JROOT/tmp/install_6682de7cc1d30/astroid-package-v3.0.15.Framework-Template/pkg_astroid.php:64
3 | pkg_astroidInstallerScript->getJoomlaVersion() | JROOT/tmp/install_6682de7cc1d30/astroid-package-v3.0.15.Framework-Template/pkg_astroid.php:31
4 | pkg_astroidInstallerScript->install() | JROOT/libraries/src/Installer/LegacyInstallerScript.php:159
5 | Joomla\CMS\Installer\LegacyInstallerScript->__call() | JROOT/libraries/src/Installer/LegacyInstallerScript.php:188
6 | Joomla\CMS\Installer\LegacyInstallerScript->callOnScript() | JROOT/libraries/src/Installer/LegacyInstallerScript.php:56
7 | Joomla\CMS\Installer\LegacyInstallerScript->install() | JROOT/libraries/src/Installer/Adapter/PackageAdapter.php:613
8 | Joomla\CMS\Installer\Adapter\PackageAdapter->triggerManifestScript() | JROOT/libraries/src/Installer/InstallerAdapter.php:777
9 | Joomla\CMS\Installer\InstallerAdapter->install() | JROOT/libraries/src/Installer/Installer.php:652
10 | Joomla\CMS\Installer\Installer->install() | JROOT/administrator/components/com_installer/src/Model/InstallModel.php:214
11 | Joomla\Component\Installer\Administrator\Model\InstallModel->install() | JROOT/administrator/components/com_installer/src/Controller/InstallController.php:52
12 | Joomla\Component\Installer\Administrator\Controller\InstallController->install() | JROOT/libraries/src/MVC/Controller/BaseController.php:730
13 | Joomla\CMS\MVC\Controller\BaseController->execute() | JROOT/libraries/src/Dispatcher/ComponentDispatcher.php:143
14 | Joomla\CMS\Dispatcher\ComponentDispatcher->dispatch() | JROOT/libraries/src/Component/ComponentHelper.php:361
15 | Joomla\CMS\Component\ComponentHelper::renderComponent() | JROOT/libraries/src/Application/AdministratorApplication.php:150
16 | Joomla\CMS\Application\AdministratorApplication->dispatch() | JROOT/libraries/src/Application/AdministratorApplication.php:195
17 | Joomla\CMS\Application\AdministratorApplication->doExecute() | JROOT/libraries/src/Application/CMSApplication.php:306
18 | Joomla\CMS\Application\CMSApplication->execute() | JROOT/administrator/includes/app.php:58
19 | require_once() | JROOT/administrator/index.php:32

Any help greatly welcome

**Comment by Whichcraft** [comment]
still I see the template installed (astroid_template_two - Default, Framework Version: 3.0.15 Template Version: 1.0.5) but when I activate it I get:

 in /var/www/html/www.XXXX/libraries/astroid/framework/library/astroid/Template.php (line 233)

            $result = $db->loadObject();
            $params = new Registry();
            $params->loadString($result->params);
            $result->params = $params;
            return $result;
        }
        protected function _getPresets()
        {

maybe some remnants of the old template/framework?

also I cannot uninstall Astroid Package, it keeps telling me Manifest file missing

**Comment by Whichcraft** [comment]
Oh, on setting the menu assignments, it at least starts to work but I get all sorts of other errors like:

https://www.XXXX/administrator/index.php?option=com_ajax&astroid=manager&id=16&t=1719853248

Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/www.XXXX/libraries/src/Log/LogEntry.php on line 126

Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 32768 bytes) in /var/www/html/wwwXXXX/libraries/vendor/symfony/error-handler/Error/OutOfMemoryError.php on line 1

**Comment by Chacapamac** [comment]
Did you try to up your **memory allocation** and **time of execution** on the server (php.ini)

**max_execution_time
max_input_time
memory_limit**

**Comment by Whichcraft** [comment]
jep, i was able to resolve php memory stuff. 

Astroid package uninstall still remaining.

I even did remove it from database xxx_extensions table and reinstalled it. Still not able to remove it again. Looks like a bug to me.

Warning
Package Uninstall: Missing manifest file.
info
Error uninstalling package.

Astroid Package | Site | Package | 3.0.15
The currently installed Joomla! version is "‎5.1.2" ‎5.1.2

**Comment by Chacapamac** [comment]
Yeah 

> Warning
> Package Uninstall: Missing manifest file.

I think I have the same warning for a long time now.
